### PR TITLE
[stable/odoo] Allow external PostgreSQL database

### LINKED ISSUE #14172: [stable/odoo] Allow external Postgres server (not the one from chart dependencies)
It would be nice if we could use Odoo chart with an existing Postgres server not configured from Odoo chart dependencies.

I have a Postgres server running in AWS RDS and I don't see how I can configure Odoo chart to use it...

See Keycloak chart as example:
https://github.com/codecentric/helm-charts/blob/master/charts/keycloak/values.yaml#L250

### PR DESCRIPTION
**Pull request by javsalgar**
Signed-off-by: Javier Salmeron Garcia <jsalmeron@vmware.com>

#### What this PR does / why we need it:
Allow using an external PostgreSQL database

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes https://github.com/helm/charts/issues/14172

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

File: stable/odoo/README.md [hunk 1]
```diff
@@ -69,6 +69,11 @@ The following table lists the configurable parameters of the Odoo chart and thei
 | `service.loadBalancer`                | Kubernetes LoadBalancerIP to request                        | `nil`                                          |
 | `service.externalTrafficPolicy`       | Enable client source IP preservation                        | `Cluster`                                      |
 | `service.nodePort`                    | Kubernetes http node port                                   | `""`                                           |
+| `externalDatabase.host`          | Host of the external database              | `localhost`                                             |
```
Comment (tompizmor) [inline stable/odoo/README.md]: Indentation, please!!! 😁

File: stable/odoo/README.md [hunk 2]
```diff
@@ -136,6 +142,17 @@ The [Bitnami Odoo](https://github.com/bitnami/bitnami-docker-odoo) image stores
 Persistent Volume Claims are used to keep the data across deployments. This is known to work in GCE, AWS, and minikube.
 See the [Configuration](#configuration) section to configure the PVC or to disable persistence.
 
+## Using an external database
+
+Sometimes you may want to have Odoo connect to an external database rather than installing one inside your cluster, e.g. to use a managed database service, or use run a single database server for all your applications. To do this, the chart allows you to specify credentials for an external database under the [`externalDatabase` parameter](#configuration). You should also disable the PostgreSQL installation with the `postgresql.enabled` option. For example:
```
Comment (tompizmor) [inline stable/odoo/README.md]: or use run a -> or use|run a

File: stable/odoo/templates/deployment.yaml [hunk 3]
```diff
@@ -26,14 +26,33 @@ spec:
         imagePullPolicy: {{ .Values.image.pullPolicy | quote }}
         env:
         - name: POSTGRESQL_HOST
+        {{- if .Values.postgresql.enabled }}
```
Comment (tompizmor) [inline stable/odoo/templates/deployment.yaml]: Instead of these if/else you could use the `ternary` operator

**Review by tompizmor — CHANGES_REQUESTED** [review]

File: stable/odoo/templates/deployment.yaml (hunk 3 above)
Comment (javsalgar) [reply stable/odoo/templates/deployment.yaml]: Done

File: stable/odoo/README.md (hunk 1 above)
Comment (javsalgar) [reply stable/odoo/README.md]: Done

File: stable/odoo/README.md (hunk 2 above)
Comment (javsalgar) [reply stable/odoo/README.md]: Done

**Review by tompizmor — APPROVED** [review]
/ok-to-test
/lgtm

**Pull request merged**